### ISSUE TITLE
Invalid parameters.

### ISSUE THREAD

**Issue by LielAmarScoRpyoN** [Commit451/YouTubeExtractor#26]
Whenever I'm trying to extract the video, I get an error saying "Invalid Parameters".

My Code:

[CODE]
private Context context;
    private YouTubeExtractor mExtractor;
    private Callback<YouTubeExtractionResult> mExtractionCallback;
    private String id;

    public Downloader(Context context, String id) {
        this.context = context;
        this.id = id;

        mExtractor = YouTubeExtractor.create();

        mExtractionCallback = new Callback<YouTubeExtractionResult>() {

            @Override
            public void onResponse(Call<YouTubeExtractionResult> call, Response<YouTubeExtractionResult> response) {
                bindVideoResult(response.body());
            }

            @Override
            public void onFailure(Call<YouTubeExtractionResult> call, Throwable t) {
                onError(t);
            }
        };
    }

    public void download() {
        mExtractor.extract("9d8wWcJLnFI").enqueue(mExtractionCallback);
        mExtractor.extract(this.id).enqueue(mExtractionCallback);
    }

    private void onError(Throwable t) {
        t.printStackTrace();
        Log.d("OnError", "id: " + id);
        Toast.makeText(context, "It failed to extract. So sad", Toast.LENGTH_SHORT).show();
    }

    private void bindVideoResult(YouTubeExtractionResult result) {
        Log.d("OnSuccess", "Got a result with the best url: " + result.getBestAvailableQualityVideoUri());
        Toast.makeText(context, "result : " + result.getSd360VideoUri(), Toast.LENGTH_SHORT).show();
    }
}
[/CODE]

**Comment by Jawnnypoo** [comment]
Looks like you are using an older version of this library. This issue is more than likely fixed and you should test with the latest to see if it is still an issue for you.

**Comment by LielAmarScoRpyoN** [comment]
What is the latest version?
By the way, I'm using 2.1.0 :)

**Comment by Jawnnypoo** [comment]
4.2.1, instructions are in the README for how to use the latest.

**Comment by LielAmarScoRpyoN** [comment]
Oh, is it only for Kothlin?

**Comment by Jawnnypoo** [comment]
It is written in Kotlin, but you can use it in Java

**Comment by LielAmarScoRpyoN** [comment]
YouTubeExtractor extractor = YouTubeExtractor.create();
YoutubeExtractor doesn't have a method called create();
YoutubeExtractor.Companion does have it though.

**Comment by Jawnnypoo** [comment]
Yep, certain calls will be different from the example when using Java instead of Kotlin

**Comment by LielAmarScoRpyoN** [comment]
After getting the result
(
```
        YouTubeExtractor extractor = YouTubeExtractor.Companion.create();
        extractor.extract("9d8wWcJLnFI").blockingGet();
```
)
How can I convert it into a downloadable?
